src/core/dbus-unit.c

```diff
@@ -1839,6 +1839,26 @@ static int bus_unit_set_live_property(
                 return 1;
         }
 
+        /* Two boolean settings that only apply to active units. We don't actually write those to /run, this
+         * state is managed internally. */
+
+        if (STR_IN_SET(name, "NeedsRestart", "NeedsReload")) {
+                bool b;
```

Comment:
sd_bus_message_read() reads an `int`, not a `bool`, this is a bit of a memory corruption...